src/Microsoft.Azure.ServiceBus/Core/ServiceBusPlugin.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.Core
+{
+    using System.Threading.Tasks;
+
+    public abstract class ServiceBusPlugin
+    {
+        public abstract string Name { get; }
```

Comment:
I would expect that this be set by the plugin, so we wouldnt want to auto generate this. We are hoping to prevent users from calling use multiple times with the same plugin.